app/models/live_requirement_template.rb

```diff
@@ -1,19 +1,22 @@
 class LiveRequirementTemplate < RequirementTemplate
-  validate :unique_classification_for_undiscarded, on: :create
+  validate :unique_classification_for_undiscarded
 
   def visibility
     "live"
   end
 
   def unique_classification_for_undiscarded
+    return unless discarded_at.nil?
+
     existing_record =
       LiveRequirementTemplate.find_by(
         permit_type_id: permit_type_id,
         activity_id: activity_id,
         first_nations: first_nations,
         discarded_at: nil
       )
-    if existing_record.present?
+
+    if existing_record.present? && existing_record.id != id
```

Comment:
I think it might be better to change this to:

```
existing_record =
      LiveRequirementTemplate.where.not(id: id).find_by(
        permit_type_id: permit_type_id,
        activity_id: activity_id,
        first_nations: first_nations,
        discarded_at: nil
      )
    if existing_record.present?

```
I think the current implementation will work, but I’m not totally sure which record find_by would return first if there are two matches. So I think the suggestion is more explicit